app/jobs/statcan_cron_job.rb

```diff
@@ -0,0 +1,14 @@
+class StatcanCronJob < ApplicationJob
+  queue_as :default
+
+  def perform(current_time = Time.current)
+    datasets = StatcanDataset.select(:id, :sync_schedule, :last_synced_at)
```

Comment:
nit: You should use a scope on for stale datasets instead of querying directly here

`datasets = StatcanDataset.stale.select(:id, :sync_schedule, :last_synced_at)`